packages/opentelemetry-core/src/utils/environment.ts

```diff
@@ -64,6 +64,26 @@ function parseNumber(
   }
 }
 
+/**
+ * Parses a LogLevelString as a LogLevel
+ * @param key
+ * @param environment
+ * @param values
+ */
+function parseLogLevel(
+  key: keyof ENVIRONMENT,
+  environment: ENVIRONMENT_MAP | ENVIRONMENT,
+  values: ENVIRONMENT_MAP
+) {
+  let logLevel: LogLevel | undefined;
+  const value = values[key];
+  if (value && typeof value === 'string') {
+    const logLevelString = value.toUpperCase() as LogLevelString;
+    logLevel = LogLevel[logLevelString];
+  }
+  environment[key] = logLevel || DEFAULT_ENVIRONMENT[key];
```

Comment:
the defaults are attached in getEnv()
here only the correct existing value should be attached

```
if (value && typeof value === 'string') {
    const logLevelString = value.toUpperCase() as LogLevelString;
    environment[key] = LogLevel[logLevelString];
  }
```

Reply:
True, the reason I did this is to deal with invalid string values.
But we can do a more thorough check that upper cased `value` is one of the LogLevelStrings

Reply:
updated!

Reply:
I was only referring to logic that default value is set in different place here you basically set the environment only if it passed all your different checks, validation etc. otherwise you don't set it and it will fallback to default automatically

Reply:
yup that was a good point, the validation was made stricter and the extraneous fall back has been removed